File: database/database-annotation-processor/src/main/java/ru/tinkoff/kora/database/annotation/processor/vertx/extension/VertxTypesExtensionFactory.java
package ru.tinkoff.kora.database.annotation.processor.vertx.extension;

import ru.tinkoff.kora.database.annotation.processor.vertx.VertxTypes;
import ru.tinkoff.kora.kora.app.annotation.processor.extension.ExtensionFactory;
import ru.tinkoff.kora.kora.app.annotation.processor.extension.KoraExtension;

import javax.annotation.processing.ProcessingEnvironment;
import java.util.Optional;

public class VertxTypesExtensionFactory implements ExtensionFactory {

    @Override
    public Optional<KoraExtension> create(ProcessingEnvironment processingEnvironment) {
        var type = processingEnvironment.getElementUtils().getTypeElement(VertxTypes.ROW_MAPPER.canonicalName());
        if (type == null) {
            return Optional.empty();
        }

        return Optional.of(new VertxTypesExtension(processingEnvironment));
    }
}

File: database/database-symbol-processor/src/main/kotlin/ru/tinkoff/kora/database/symbol/processor/vertx/extension/VertxTypesExtensionFactory.kt
package ru.tinkoff.kora.database.symbol.processor.vertx.extension

import com.google.devtools.ksp.processing.CodeGenerator
import com.google.devtools.ksp.processing.KSPLogger
import com.google.devtools.ksp.processing.Resolver
import ru.tinkoff.kora.kora.app.ksp.extension.ExtensionFactory

class VertxTypesExtensionFactory: ExtensionFactory {
    override fun create(resolver: Resolver, kspLogger: KSPLogger, codeGenerator: CodeGenerator) = VertxTypesExtension(resolver, kspLogger, codeGenerator)
}
